File: backend/models/audit_report.py
from dataclasses import dataclass, field
from typing import List


@dataclass
class AuditReportRecord:
    report_id: str
    shift_id: str
    generated_at: str         # ISO 8601
    patrol_summary: str
    alert_summary: str
    rag_examples_used: List[str] = field(default_factory=list)
    report_text: str = ""
    model_used: str = "gemini-2.5-flash"

File: frontend/types/auditReport.ts
export interface AuditReportRecord {
  report_id: string;
  shift_id: string;
  guard_id: string;
  generated_at: string;     // ISO 8601
  patrol_summary: string;
  alert_summary: string;
  rag_examples_used: string[];
  report_text: string;
  model_used: string;
}
